bin/oneoff/import_unit_details.rb

```diff
@@ -82,22 +91,104 @@ def main(options)
 
     cb_unit = JSON.parse(cb_unit_json)
     validate_unit(script, cb_unit)
+    chapters = get_validated_chapters(cb_unit)
+    lesson_pairs = get_validated_lesson_pairs(script, chapters)
+
+    log "validated unit #{script.name} data"
+
+    next if options.dry_run
+
+    lesson_pairs.each do |lesson, cb_lesson|
+      lesson.update_from_curriculum_builder(cb_lesson)
+    end
+    log "updated #{lesson_pairs.count} lessons in unit #{script.name}"
   end
 end
 
 def fetch(url)
-  log "fetching unit json from #{url}"
   uri = URI(url)
   response = Net::HTTP.get_response(uri)
   raise "HTTP status #{response.code} fetching #{uri}" unless response.is_a? Net::HTTPSuccess
   body = response.body
-  log "received #{body.length} bytes of unit json."
+  log "fetched #{body.length} bytes of unit json from #{uri}"
   body
 end
 
 def validate_unit(script, cb_unit)
   raise "unexpected unit_name #{cb_unit['unit_name']}" unless cb_unit['unit_name'] == script.name
-  log "validated unit data for #{script.name}"
+end
+
+def get_validated_chapters(cb_unit)
+  # In 2020, CSF and CSD lessons are all inside chapters, and CSP does not use
```

Comment:
Hey Dave - I looks like CSP 2020 uses chapters on the LB side. https://levelbuilder-studio.code.org/s/csd1-2020

Reply:
Hey Dani, that looks like a link to CSD, but your point is still valid -- CSP 2020 does use lesson groups on the LB side. I am using "chapter" to refer to data coming in from CB, and what I am trying to say here is that CSP 20 does not use Chapters on CB.

It would probably help if I clarified when I am talking about CB data without relying on the assumption that "chapter" means CB. I will update this in the next PR.

Reply:
Sounds good. I just wanted to check. (Sorry about the wrong link lol)

Reply:
Addressed in https://github.com/code-dot-org/code-dot-org/pull/37772/commits/1c167d9ffc0adaa47c4f037cf195fec4b01dee79